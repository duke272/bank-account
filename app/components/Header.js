import React from 'react';
import PropTypes from 'prop-types';
import { AppBar, Typography } from '@material-ui/core';
import EuroIcon from '@material-ui/icons/Euro';
import IconButton from '@material-ui/core/IconButton';
import SortIcon from '@material-ui/icons/Sort';
import { formatAmount } from '../containers/AccountOverview/helpers';

// @TODO don't display 0 as default balance

const Header = ({ amount, debitCredit, sortOrder, toggleSort }) => (
  <AppBar
    position="fixed"
    style={{ padding: '8px 8px 8px 16px', flexDirection: 'row' }}
  >
    <Typography variant="h6" style={{ flexGrow: 1 }}>
      Account Balans: <EuroIcon fontSize="inherit" />
      {formatAmount(amount, debitCredit)}
    </Typography>
    <IconButton
      onClick={toggleSort}
      style={{
        flexGrow: 1,
        padding: 0,
        color: 'inherit',
        transform: sortOrder === 'asc' ? 'scaleX(-1)' : 'none',
      }}
    >
      <SortIcon />
    </IconButton>
  </AppBar>
);

Header.propTypes = {
  amount: PropTypes.number,
  debitCredit: PropTypes.oneOf(['credit', 'debit']),
  sortOrder: PropTypes.oneOf(['asc', 'desc']),
  toggleSort: PropTypes.func,
};

export default Header;
