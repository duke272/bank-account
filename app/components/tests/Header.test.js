import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';

describe('<Header />', () => {
  it('should render without data', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a negative amount', () => {
    const tree = renderer
      .create(<Header amount={124.77} debit_credit="credit" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a positive amount', () => {
    const tree = renderer
      .create(<Header amount={55.66} debit_credit="debit" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
