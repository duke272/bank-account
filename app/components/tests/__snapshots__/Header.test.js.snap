// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render a negative amount 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  style={
    Object {
      "flexDirection": "row",
      "padding": "8px 8px 8px 16px",
    }
  }
>
  <h6
    className="MuiTypography-root MuiTypography-h6"
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    Account Balans: 
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5H15l1-2H8.58c-.05-.33-.08-.66-.08-1s.03-.67.08-1H15l1-2H9.24C10.32 6.92 12.5 5.5 15 5.5c1.61 0 3.09.59 4.23 1.57L21 5.3C19.41 3.87 17.3 3 15 3c-3.92 0-7.24 2.51-8.48 6H3l-1 2h4.06c-.04.33-.06.66-.06 1s.02.67.06 1H3l-1 2h4.52c1.24 3.49 4.56 6 8.48 6 2.31 0 4.41-.87 6-2.3l-1.78-1.77c-1.13.98-2.6 1.57-4.22 1.57z"
      />
    </svg>
    -124,77
  </h6>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "color": "inherit",
        "flexGrow": 1,
        "padding": 0,
        "transform": "none",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h6v-2H3v2zM3 6v2h18V6H3zm0 7h12v-2H3v2z"
        />
      </svg>
    </span>
  </button>
</header>
`;

exports[`<Header /> should render a positive amount 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  style={
    Object {
      "flexDirection": "row",
      "padding": "8px 8px 8px 16px",
    }
  }
>
  <h6
    className="MuiTypography-root MuiTypography-h6"
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    Account Balans: 
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5H15l1-2H8.58c-.05-.33-.08-.66-.08-1s.03-.67.08-1H15l1-2H9.24C10.32 6.92 12.5 5.5 15 5.5c1.61 0 3.09.59 4.23 1.57L21 5.3C19.41 3.87 17.3 3 15 3c-3.92 0-7.24 2.51-8.48 6H3l-1 2h4.06c-.04.33-.06.66-.06 1s.02.67.06 1H3l-1 2h4.52c1.24 3.49 4.56 6 8.48 6 2.31 0 4.41-.87 6-2.3l-1.78-1.77c-1.13.98-2.6 1.57-4.22 1.57z"
      />
    </svg>
    -55,66
  </h6>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "color": "inherit",
        "flexGrow": 1,
        "padding": 0,
        "transform": "none",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h6v-2H3v2zM3 6v2h18V6H3zm0 7h12v-2H3v2z"
        />
      </svg>
    </span>
  </button>
</header>
`;

exports[`<Header /> should render without data 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  style={
    Object {
      "flexDirection": "row",
      "padding": "8px 8px 8px 16px",
    }
  }
>
  <h6
    className="MuiTypography-root MuiTypography-h6"
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    Account Balans: 
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5H15l1-2H8.58c-.05-.33-.08-.66-.08-1s.03-.67.08-1H15l1-2H9.24C10.32 6.92 12.5 5.5 15 5.5c1.61 0 3.09.59 4.23 1.57L21 5.3C19.41 3.87 17.3 3 15 3c-3.92 0-7.24 2.51-8.48 6H3l-1 2h4.06c-.04.33-.06.66-.06 1s.02.67.06 1H3l-1 2h4.52c1.24 3.49 4.56 6 8.48 6 2.31 0 4.41-.87 6-2.3l-1.78-1.77c-1.13.98-2.6 1.57-4.22 1.57z"
      />
    </svg>
    -NaN
  </h6>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "color": "inherit",
        "flexGrow": 1,
        "padding": 0,
        "transform": "none",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h6v-2H3v2zM3 6v2h18V6H3zm0 7h12v-2H3v2z"
        />
      </svg>
    </span>
  </button>
</header>
`;
